### PR TITLE
test: add benchmark for log.fromEntryHash()

### PR DESCRIPTION
**Pull request by mistakia**
Let me know if you prefer consolidating these benchmarks to one PR.

Also, let me know if you have any suggestions on how to approach memory heap usage profiling. 

Related: orbitdb/ipfs-log#136

### CODE REVIEW

File: benchmarks/benchmark-from-entry-hash.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    console.log("Creating a log...")
+
+    const st = new Date().getTime()
+
+    try {
+    for (let i = 1; i < count + 1; i ++) {
+      await log.append('hello' + i, refCount)
+      process.stdout.write("\rWriting " + i + " / " + count)
+    }
+    const dt1 = new Date().getTime()
```
Comment (haadcode) [inline benchmarks/benchmark-from-entry-hash.js]: Would be nice to align the indentation here and the lines above to match the try/catch scope.

File: benchmarks/benchmark-from-entry-hash.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+      -1,
+      [],
+      log._key,
+      log._keys,
+      onDataUpdated
+    )
+
+    // total = result.length
+    // queriesPerSecond = result.length
+    // queriesPerSecond = totalQueries - queriesPerSecond
```
Comment (haadcode) [inline benchmarks/benchmark-from-entry-hash.js]: Before merging, we should remove all the commented lines.

File: benchmarks/benchmark-from-entry-hash.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    outputMetrics()
+    const et = new Date().getTime()
+    console.log("Loading took:", (et - dt2), "ms")
+    const memory = process.memoryUsage()
+    console.log(`Memory Heap Used: ${memory.heapUsed / 1024 / 1024} MB`)
+    console.log(`Memory Heap Total: ${memory.heapTotal / 1024 / 1024} MB`)
```
Comment (haadcode) [inline benchmarks/benchmark-from-entry-hash.js]: Very cool to output the memory usage! 👍

File: benchmarks/benchmark-from-entry-hash.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    const outputMetrics = () => {
+      // queriesPerSecond = total - queriesPerSecond
+      totalQueries = total - totalQueries
+      seconds++
+      if (seconds % 10 === 0) {
+        console.log(`--> Average of ${lastTenSeconds / 10} q/s in the last 10 seconds`)
+        if (lastTenSeconds === 0) throw new Error('Problems!')
+        lastTenSeconds = 0
+      }
+      console.log(`\n${queriesPerSecond} queries per second, ${totalQueries} queries in ${seconds} seconds (Entry count: ${total})`)
```
Comment (haadcode) [inline benchmarks/benchmark-from-entry-hash.js]: If I read the code correctly, we're benchmarking "how long does it take to load a log" and as such we're not necessarily doing queries, but "entries per second". I feel it'd be worthwhile to make the output messages clearer on that and display something like "X entries loaded per second" etc.

What do you think?

If you feel that would improve the PR, we should also rename the variables used in the code to match the output, eg. `queriesPerSecond` -> `entriesLoadedPerSecond` or something similar.

File: benchmarks/benchmark-from-entry-hash.js (hunk 3 above)
Comment (haadcode) [reply benchmarks/benchmark-from-entry-hash.js]: Would it make sense to snapshot/capture the heap used also before the benchmark is run and do a delta comparison (ie. what is the differences between before and after)?

File: benchmarks/benchmark-from-entry-hash.js (hunk 4 above)
Comment (mistakia) [reply benchmarks/benchmark-from-entry-hash.js]: No brainer change - lazy of me to have not done it before 😁

**Comment by haadcode** [comment]
Thank you @mistakia, this looks good to me 👍

**Pull request merged**